prefect.yaml

```
@@ -1,35 +1,57 @@
-# Welcome to your prefect.yaml file! You can use this file for storing and managing
-# configuration for deploying your flows. We recommend committing this file to source
-# control along with your flow code.
+name: prefect-docker-pixi-tiled-smi
+prefect-version: 3.6.0
 
-# Generic metadata about this project
-name: workflows
-prefect-version: 2.18.1
-
-# build section allows you to manage and build docker images
 build:
+  - prefect.deployments.steps.run_shell_script:
+      id: get-commit-hash
+      script: git rev-parse --short HEAD
+      stream_output: false
+  - prefect_docker.deployments.steps.build_docker_image:
+      id: build-image
+      requires: prefect-docker>0.6.0
+      image_name: ghcr.io/nsls2/smi-workflow-prefect3
+      tag: "{{ get-commit-hash.stdout }}"
+      dockerfile: Dockerfile
+      additional_tags:
+        - main
 
-# push section allows you to manage if and how this project is uploaded to remote locations
 push:
+  - prefect_docker.deployments.steps.push_docker_image:
+      requires: prefect-docker>0.6.0
+      image_name: "{{ build-image.image_name }}"
+      tag: "{{ build-image.tag }}"
+      credentials:
+        "{{
+        prefect.blocks.docker-registry-credentials.ghcr-nsls2-smi-workflow-prefect3
+        }}"
+      additional_tags:
+        - main
 
-# pull section allows you to provide instructions for cloning this project in remote locations
 pull:
-- prefect.deployments.steps.git_clone:
-    repository: https://github.com/NSLS-II-SMI/workflows
-    branch: main
+  - prefect.deployments.steps.set_working_directory:
+      directory: /repo
+  - prefect.deployments.steps.git_clone:
+      repository: https://github.com/JunAishima/smi-workflows.git
+      branch: prefect3-pixi-upgrade
 
-# the deployments section allows you to provide configuration for deploying flows
 deployments:
-- name: smi-end-of-run-workflow
-  version:
-  tags:
-  - smi
-  description:
-  entrypoint: end_of_run_workflow.py:end_of_run_workflow
-  parameters: {}
-  work_pool:
-    name: smi-work-pool
-    work_queue_name:
-    job_variables: {}
-  schedule:
-  is_schedule_active: true
\ No newline at end of file
+  - name: smi-end-of-run-workflow-docker
+    version: 0.1.1
```

Additional tags for beamline/endstation and branch to keep things consistent.
```suggestion
    version: 0.1.1
    tags:
      - smi
      - main
```